servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -1429,6 +1436,78 @@ Size2 TextureStorage::texture_size_with_proxy(RID p_proxy) {
 	return texture_2d_get_size(p_proxy);
 }
 
+void TextureStorage::texture_rd_initialize(RID p_texture, const RID &p_rd_texture) {
+	ERR_FAIL_COND(!RD::get_singleton()->texture_is_valid(p_rd_texture));
+
+	// TODO : investigate if we can support this, will need to be able to obtain the order and obtain the slice info
+	ERR_FAIL_COND_MSG(RD::get_singleton()->texture_is_shared(p_rd_texture), "Please create the texture object using the original texture");
+
+	RD::TextureFormat tf = RD::get_singleton()->texture_get_format(p_rd_texture);
+	ERR_FAIL_COND(!(tf.usage_bits & RD::TEXTURE_USAGE_SAMPLING_BIT));
+
+	TextureFromRDFormat imfmt;
+	_texture_format_from_rd(tf.format, imfmt);
+	ERR_FAIL_COND(imfmt.image_format == Image::FORMAT_MAX);
+
+	Texture texture;
+
+	switch (tf.texture_type) {
+		// Q? What to do with _1D and cube textures?
```

Comment:
RS doesn't do anything with 1D textures, so we can ignore them for now. But this needs to support CUBEMAP and CUBEMAP_ARRAY as they are supported by ``texture_2d_layered_initialize``

Reply:
the problem is that on the RD side we don't identify them as such, it's just a 2D texture with 6 layers for CUBEMAP and a multiple of 6 for CUBEMAP_ARRAY. Take a look at the TextureCubemapRD implementation I did.

Reply:
@clayjohn `texture_2d_layered_create` isn't used here. You mean adding the parameter to `texture_rd_create`/`texture_rd_initialize` ?

Reply:
@clayjohn let me know what you think of the changes I made

Reply:
Looks good! You can delete the comment now
```suggestion
```